common/src/types.rs

```diff
@@ -71,18 +71,20 @@ pub struct AgendaProof {
     pub proof: Vec<TypedSignature<Agenda>>,
 }
 
+/// It holds the reserved state as a `Box` to flatten the variant size.
+/// (see https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant)
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 pub enum Diff {
     /// Nothing changed in the repository; an empty commit.
     None,
-    /// Only changes non-reserved areas. Contains the hash of the diff.
+    /// Changes the reserved area ONLY.
+    Reserved(Box<ReservedState>),
+    /// General diff that may change both the reserved state and the others.
```

Comment:
```suggestion
    /// General diff that may change both the reserved state and the non-reserved state.
```
how about making it clear like this?

Reply:
fixed